src/functionalTest/kotlin/kotlinx/kover/test/functional/core/Runner.kt

```diff
@@ -1,326 +1,108 @@
+    val pluginClasspath = pluginClasspath + classpathFile.readLines().map { File(it) }
+    withPluginClasspath(pluginClasspath)
+}
 
-        if (engine == CoverageEngine.INTELLIJ) {
-            builder.appendLine("    coverageEngine.set(kotlinx.kover.api.CoverageEngine.INTELLIJ)")
-            if (intellijVersion != null) {
-                builder.appendLine("""    intellijEngineVersion.set("$intellijVersion")""")
-            }
-        }
-        if (engine == CoverageEngine.JACOCO) {
-            builder.appendLine("    coverageEngine.set(kotlinx.kover.api.CoverageEngine.JACOCO)")
-            if (jacocoVersion != null) {
-                builder.appendLine("""    jacocoEngineVersion.set("$jacocoVersion")""")
-            }
-        }
 
-        koverConfigs.forEach {
-            builder.appendLine("    $it")
-        }
-        builder.appendLine("}")
+private class RunResultImpl(private val result: BuildResult, dir: File) : RunResult {
+    val buildDir: File = File(dir, "build")
 
-        return builder.toString()
+    override fun output(checker: String.() -> Unit) {
+        result.output.checker()
     }
 
-    @Suppress("UNUSED_PARAMETER")
-    private fun buildVerifications(language: GradleScriptLanguage, verifications: List<VerificationRule>): String {
-        return ""
+    override fun file(name: String, checker: File.() -> Unit) {
+        File(buildDir, name).checker()
     }
 
-    private fun buildSettings(language: GradleScriptLanguage): String {
-        return settingsTemplate(language)
-            .replace("//SUBMODULES", buildSubmodulesIncludes(submodules.keys))
+    override fun xml(filename: String, checker: XmlReport.() -> Unit) {
+        val xmlFile = File(buildDir, filename)
+        if (!xmlFile.exists()) throw IllegalStateException("XML file '$filename' not found")
+        XmlReportImpl(xmlFile).checker()
     }
 
-    private fun buildScripts(language: GradleScriptLanguage): String {
-        val scripts = if (language == GradleScriptLanguage.KOTLIN) kotlinScripts else groovyScripts
-
-        return if (scripts.isNotEmpty()) {
-            scripts.joinToString("\n", "\n", "\n")
-        } else {
-            ""
-        }
+    override fun outcome(taskPath: String, checker: TaskOutcome.() -> Unit) {
+        result.task(taskPath)?.outcome?.checker()
+            ?: throw IllegalArgumentException("Task '$taskPath' not found in build result")
     }
 }
 
-private data class ProjectSlice(val language: GradleScriptLanguage, val type: ProjectType, val engine: CoverageEngine)
 
-private class SubmoduleBuilderImpl : ModuleBuilder<SubmoduleBuilderImpl> {
-    override fun verification(rules: Iterable<VerificationRule>): SubmoduleBuilderImpl {
-        TODO("Not yet implemented")
-    }
+private class XmlReportImpl(file: File) : XmlReport {
+    private val document = DocumentBuilderFactory.newInstance()
+        .also { it.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false) }
```

Comment:
The option with this name disables checking the dtd file, because the JaCoCo report has the `<!DOCTYPE report PUBLIC "-//JACOCO//DTD Report 1.1//EN" "report.dtd">`

Reply:
Then it worth commenting, I think